### PR TITLE
Features/log in page

### PR DESCRIPTION
**Pull request by scokljat**
## Changes

-Use "vue3-google-oauth2" package for authentication.
-Add authentication service.
-Edit css style for logged and unlogged pages.

## Reference
https://trello.com/c/uNZDujyz/48-task-features-loginpage

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -2,10 +2,17 @@ import { createApp } from 'vue';
...
-
+import GAuth from 'vue3-google-oauth2';
 // table imports
 import 'element-plus/es/components/table/style/css';
 import 'element-plus/es/components/table-column/style/css';
 import 'element-plus/es/components/pagination/style/css';
 
-createApp(App).use(store).use(router).mount('#app');
+const gAuthOptions = {
+  clientId: '808924355914-8k4928tljobpmqc276b5iqa60fshkubi',
```
Comment (thedev966) [inline src/main.js:12]: Use .env variable instead.

File: src/pages/LoginPage.vue [hunk 2]
```diff
@@ -1,3 +1,74 @@
 <template>
-  <h1>Log in Page</h1>
+  <div class="container">
+    <h1>Welcome Bot</h1>
+    <h2>Log in</h2>
+    <button @click="handleClickSignIn">
```
Comment (thedev966) [inline src/pages/LoginPage.vue:5]: Maybe name it `handleSignIn` instead?

File: src/store/index.js [hunk 3]
```diff
@@ -13,7 +16,17 @@ export default createStore({
     [CLOSE_APP_MODAL]: (state) => {
       state.isModalActive = false;
     },
+    [SET_USER]: (state, { token }) => {
+      state.isLoggedIn = true;
+      state.token = { token };
```
Comment (thedev966) [inline src/store/index.js:21]: `state.token = token`

File: src/api/services/LogInService.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const AuthService = {
```
Comment (josip2312) [inline src/api/services/LogInService.js:1]: Your file name and service name don't match

File: src/main.js (hunk 1 above)
Comment (josip2312) [reply src/main.js:12]: Agreed

File: src/pages/LoginPage.vue (hunk 2 above)
Comment (josip2312) [reply src/pages/LoginPage.vue]: Agreed

File: src/pages/LoginPage.vue [hunk 5]
```diff
@@ -1,3 +1,74 @@
...
+    @include tablet-md {
+      font-size: 1.9rem;
+    }
+  }
+}
+
+button {
+  background-color: var(--primary);
+  width: 9.4rem;
+  height: 4.4rem;
```
Comment (josip2312) [inline src/pages/LoginPage.vue:62]: This does not seem right, a fixed width and height should be set very rarely on an element

**Review by thedev966 — APPROVED** [review]

**Review by josip2312 — CHANGES_REQUESTED** [review]

**Comment by josip2312** [comment]
Address all the recommended changes

**Review by josip2312 — APPROVED** [review]

**Pull request merged**